relayer/relayer/src/contracts/azero.rs

```diff
@@ -172,27 +135,36 @@ impl MostInstance {
     pub async fn is_halted(&self, connection: &Connection) -> Result<bool, AzeroContractError> {
         Ok(self
             .contract
-            .contract_read0::<Result<bool, _>, _>(connection, "is_halted")
+            .read0::<Result<bool, _>, _>(connection, "is_halted", Default::default())
             .await??)
     }
 
-    pub async fn _needs_signature(
+    pub async fn needs_signature(
         &self,
         connection: &Connection,
         request_hash: [u8; 32],
         account: AccountId,
         committee_id: u128,
+        block: BlockStatus,
```

Comment:
I don't see a situation where we would need the make a query at a block other than the finalized.